File: c/Grace.c
#include <stdio.h>
#include <fcntl.h>
/*
	This program will print its own source when run.
*/
#define CHILD "Grace_kid.c"
#define CODE(x) int x(){int fd = creat(CHILD, 0600); char str[]="#include <stdio.h>%1$c#include <fcntl.h>%1$c/*%1$c	This program will print its own source when run.%1$c*/%1$c#define CHILD %2$cGrace_kid.c%2$c%1$c#define CODE(x) int x(){int fd = creat(CHILD, 0600); char str[]=%2$c%3$s%2$c; dprintf(fd, str, 10, 34, str); return 0;}%1$c#define MAIN main%1$c%1$cCODE(MAIN)%1$c"; dprintf(fd, str, 10, 34, str); return 0;}
#define MAIN main

CODE(MAIN)

File: python/Grace.py
#!/usr/bin/python3

"""
	This program will print its own source when run.
"""

mystr = "#!/usr/bin/python3{0}{0}{1}{1}{1}{0}	This program will print its own source when run.{0}{1}{1}{1}{0}{0}mystr = {1}{2}{1}{0}{0}with open({1}Grace_kid.py{1}, {1}w{1}) as f:{0}    f.write(mystr.format(chr(10), chr(34), mystr)){0}"

with open("Grace_kid.py", "w") as f:
    f.write(mystr.format(chr(10), chr(34), mystr))
